File: csharp/tracks-on-tracks-on-tracks/TracksOnTracksOnTracks.cs
using System;
using System.Collections.Generic;

public static class Languages
{
    public static List<string> NewList() => new List<string>();

    public static List<string> GetExistingLanguages() => new List<string> { "C#", "Clojure", "Elm" };

    public static List<string> AddLanguage(List<string> languages, string language)
    {
        languages.Add(language);

        return languages;
    }

    public static int CountLanguages(List<string> languages) => languages.Count;

    public static bool HasLanguage(List<string> languages, string language) => languages.Contains(language);

    public static List<string> ReverseList(List<string> languages)
    {
        languages.Reverse();

        return languages;
    }

    public static bool IsExciting(List<string> languages)
    {
        throw new NotImplementedException("Please implement the static Languages.IsExciting() method");
    }

    public static List<string> RemoveLanguage(List<string> languages, string language)
    {
        throw new NotImplementedException("Please implement the static Languages.RemoveLanguage() method");
    }

    public static bool IsUnique(List<string> languages)
    {
        throw new NotImplementedException("Please implement the static Languages.IsUnique() method");
    }
}

File: fsharp/tracks-on-tracks-on-tracks/TracksOnTracksOnTracks.fs
module TracksOnTracksOnTracks

let newList: string list = []

let existingList: string list = ["F#"; "Clojure"; "Haskell"]

let addLanguage (language: string) (languages: string list): string list = language :: languages

let countLanguages (languages: string list): int = languages.Length

let reverseList(languages: string list): string list = failwith "Please implement the 'reverseList' function"

let excitingList (languages: string list): bool = failwith "Please implement the 'excitingList' function"
